synapse/types.py

```diff
@@ -625,6 +625,22 @@ async def to_string(self, store: "DataStore") -> str:
             return "s%d" % (self.stream,)
 
 
+class StreamKeyType:
+    """Known stream types.
+
+    A stream is a list of entities ordered by an incrementing "stream token".
+    """
+
+    ROOM = "room_key"
+    PRESENCE = "presence_key"
+    TYPING = "typing_key"
+    RECEIPT = "receipt_key"
+    ACCOUNT_DATA = "account_data_key"
+    PUSH_RULES = "push_rules_key"
+    TO_DEVICE = "to_device_key"
+    DEVICE_LIST = "device_list_key"
```

Comment:
Thanks for the context! Yes, it looks like `StrEnum` has the drawback of interacting poorly with our string interning code.

Seeing as these types are fairly pervasive throughout the codebase, I would advocate for `Final` just to avoid hard-to-debug surprises.